assets/js/blocks/product-collection/inspector-controls/upgrade-notice.tsx

```diff
@@ -30,27 +35,44 @@ const buttonLabel = __(
 );
 
 type UpgradeNoticeProps = {
+	isSelected: boolean;
 	revertMigration: () => void;
 };
 
 const UpgradeNotice = ( { revertMigration }: UpgradeNoticeProps ) => {
 	const [ upgradeNoticeStatus, setUpgradeNoticeStatus ] =
 		useLocalStorageState(
 			MIGRATION_STATUS_LS_KEY,
-			INITIAL_STATUS_LS_VALUE
+			getInitialStatusLSValue()
 		);
+
+	const [ canCountDisplays, setCanCountDisplays ] = useState( true );
```

Comment:
Great, I really appreciate your straightforward approach! 🚀 Just wondering, would it make sense to opt for `useRef` rather than `useState` in this context?

Reply:
Great suggestion, we can avoid unnecessary re-render 👍 changed that to `useRef`. I re-tested and all seems to be working just fine.

https://github.com/woocommerce/woocommerce-blocks/pull/10494/commits/30197cfaf0ae630fd096d85797b6563a6d0084a2